### ISSUE TITLE
Pulling & unpacking some images fails with native snapshotter

### ISSUE THREAD

**Issue by dalbani** [canonical/microk8s#3499]
#### Summary
Pulling & unpacking some Docker images fails when using the `native` snapshotter.
(The alternative of using the `overlayfs`, where the problem doesn't materialize, is not an option for me, as I'm running on ZFS. As explained on https://github.com/canonical/microk8s/pull/1182/files.)

#### What Should Happen Instead?
It shouldn't have failed or generated any errors.

#### Reproduction Steps

1. Run:
   ```
   microk8s.ctr images pull --snapshotter=native docker.io/percona/percona-server-mongodb:4.2.22-22
   ...
   unpacking linux/amd64 sha256:da4634df780563e10a547662c58c8ce28fbe5c98e1ac8b42b4f6be87f292e92b...
   ctr: failed to prepare extraction snapshot "extract-596942709-j27y 
   sha256:95b37b87b7f31fe12bdd393b27f46521fa8e1cda38924ecc52c6801731feba62": copying of parent failed: unsupported mode prw-------: %!w(<nil>): unknown
   ```

#### Can you suggest a fix?
Upgrade to a version of `containerd` which includes `continuity` v0.3.0 having the following fix: https://github.com/containerd/continuity/commit/def67296172f65f5827e5355efac79e0c1331a48.
Related bug: https://github.com/containerd/containerd/issues/7054.
Unfortunately, it looks like the `continuity` upgrade hasn't been included in any release of `containerd` yet: https://github.com/containerd/containerd/commit/088c8df3ac905f90c10e7ed91be3fbb313318b33.

**Comment by dalbani** [comment]
In the meantime, I'm using the `zfs` snapshotter, although it wasn't really a recommended option from what I read in #1378.

What's the proper way to configure such a change by the way: I resorted to edit `/var/snap/microk8s/current/args/containerd-template.toml` and change `snapshotter = "${SNAPSHOTTER}"` for `snapshotter = "zfs"`.
As it wasn't clear to me how to set this `${SNAPSHOTTER}` variable.
Is there a better/cleaner way?

**Comment by dalbani** [comment]
I've just seen that `containerd` has recently be released at version 1.6.9 and it contains an updated version of `continuity`.
See https://github.com/containerd/containerd/releases/tag/v1.6.9

**Comment by dalbani** [comment]
Now that https://github.com/canonical/microk8s/pull/3536 has been merged, I have tested with the `1.25/edge` channel and revision 4193 in particular.
After having switched the snapshotter from `zfs` back to `native`, I can confirm that the error reported by containerd has now disappeared. ✌️
